src/geometry/manifold/algebra/monoid.lean

```diff
@@ -19,28 +17,24 @@ semigroups.
 -/
 
 section
-set_option old_structure_cmd true
 
 /-- Basic hypothesis to talk about a smooth (Lie) additive monoid or a smooth additive
 semigroup. A smooth additive monoid over `α`, for example, is obtained by requiring both the
 instances `add_monoid α` and `has_smooth_add α`. -/
-@[ancestor smooth_manifold_with_corners]
 class has_smooth_add {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
   {H : Type*} [topological_space H]
   {E : Type*} [normed_group E] [normed_space 𝕜 E] (I : model_with_corners 𝕜 E H)
-  (G : Type*) [has_add G] [topological_space G] [has_continuous_add G] [charted_space H G]
-  extends smooth_manifold_with_corners I G : Prop :=
```

Comment:
Can you explain the decision for not extending `smooth_manifold_with_corners`? When you will want to declare a Lie group, say, it means you will need one more `variables` entry. Moreover, I don't think one will ever use seriously `has_smooth_add` without also assuming that one has a smooth manifold with corners, no?

Reply:
I wonder if the bundled approach will lead to some diamonds later. E.g., you can't say `[has_smooth_mul G] [lie_add_group G]`. Of course, we have typeclasses for smooth (semi)rings but what if we'll decide to formalize Riemannian manifolds as a typeclass (I'm not sure that it's a good idea - sometimes you want to deal with the space of all Riemannian metrics on a manifold)? Then we'll have to add `[Riemannian_lie_group]` etc.

Reply:
I am not convinced here. The classes have exactly the same parameters, so creating a superclass makes it possible to avoid one entry in `variables`. And I am not afraid to have to add a few more classes -- this is what we do in algebra, where it seems to work pretty well, and I don't think we will need more than two or three of them. Why not going back to the previous approach, and changing it in the future if needed?

Reply:
My argument is invalid for a simple reason: `smooth_manifold_with_corners` is in `Prop`, so we can't have any diamonds. I'll go back to extending `smooth_manifold_with_corners`.